test/merge-patient-lab-iips-mocha.js

```diff
@@ -5,151 +5,260 @@ var chaiAsPromised = require('chai-as-promised');
-                    labwork.disconnect();
-                    component.inPorts.patient.detach(labwork);
-                    return component;
-                }).then(promiseLater).then(function(component){
-                    return component.rpf.createVni('').getInputState('labwork', '', '');
-                }).then(_.keys).then(_.sortBy).should.become(_.sortBy(['data','lm', 'previousLms', 'selfPort']));
+
+            describe('#errorState', function() {
+
+                it("should initially return nothing", function() {
+                    return Promise.resolve(componentFactory.getComponent )
+                        .then(commonTest.createComponent).then(function(component){
+                            return component.rpf.vni('').errorState();
+                    }).should.become(undefined);
+                });
             });
-            it("should have patient input state data after input", function() {
-                return Promise.resolve(createComponent()).then(function(component){
-                    var labwork = noflo.internalSocket.createSocket();
-                    component.inPorts.labwork.attach(labwork);
-                    labwork.send({id: '001',  glucose: '75',  date: '2012-02-01'});
-                    labwork.disconnect();
-                    component.inPorts.patient.detach(labwork);
-                    return component;
-                }).then(promiseLater).then(function(component){
-                    return component.rpf.createVni('').getInputState('labwork', '', '');
-                }).then(_.property('data')).should.become({id: '001',  glucose: '75',  date: '2012-02-01'});
+
+            describe('#inputState', function() {
+
+                it("should initially return nothing", function() {
+                    return Promise.resolve(componentFactory.getComponent )
+                        .then(commonTest.createComponent).then(function(component){
+                            return component.rpf.vni('').inputState('patient');
+                    }).should.become(undefined);
+                });
+
+                it("should have input state after input", function() {
+                    return Promise.resolve(componentFactory.getComponent )
+                        .then(commonTest.createComponent).then(function(component){
+
+                            // initialize state
+                            component.rpf.vni('').inputState( 'patient',
+                                                              { id: '001', 
+                                                                name: 'Alice',
+                                                                dob: '1979-01-23' } );
+
+                            var currentState = component.rpf.vni('').inputState('patient');
+                            currentState.should.be.an('object');
+                            currentState.should.have.all.keys( 'data', 'lm', 'port' );
```

Comment:
We can leave off the 'port' property for this sprint, since we are not using it yet.